controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java

```diff
@@ -346,20 +347,48 @@ else if (timedOut(id, deployment.get(), endpointTimeout)) {
                 return Optional.of(error);
             }
         }
-        controller.jobController().locked(id, lockedRun -> lockedRun.withSummary(summary));
 
-        if (timedOut(id, deployment.get(), installationTimeout)) {
+        boolean failed = false;
+
+        NodeList suspendedTooLong = nodeList.suspendedSince(controller.clock().instant().minus(installationTimeout));
+        if ( ! suspendedTooLong.isEmpty()) {
+            logger.log(INFO, "Some nodes have been suspended for more than " + installationTimeout.toMinutes() + " minutes.");
```

Comment:
Instead of "Some nodes" - write the list or a sample node? (btw, I think you should write a util to convert a list of hosts to String, and write the full list if the number of elements is below some reasonable threshold, while otherwise write a prefix of the list and then something like ... (and X more hosts). You could use this util here.)

Reply:
Would the number of allowed down hosts ever exceed this threshold? For the start and end (with failure), I think all ghosts should be printed.

Reply:
Possibly - I'm just thinking such a util might be used to dump out a lot of hosts - like all hosts or all but a few - but if that's not possible there is no point is making it complicated.